src/tableproperties/tablepropertiesui.js

```diff
@@ -0,0 +1,280 @@
+		const editor = this.editor;
+		const viewDocument = editor.editing.view.document;
+		const view = new TablePropertiesView( editor.locale );
+
+		// Render the view so its #element is available for the clickOutsideHandler.
+		view.render();
+
+		this.listenTo( view, 'submit', () => {
+			this._hideView();
+		} );
+
+		this.listenTo( view, 'cancel', () => {
+			editor.execute( 'undo', this._undoStepBatch );
+			this._hideView();
+		} );
+
+		// Close the balloon on Esc key press.
+		view.keystrokes.set( 'Esc', ( data, cancel ) => {
+			this._hideView();
+			cancel();
+		} );
+
+		// Reposition the balloon or hide the form if a table is no longer selected.
+		this.listenTo( editor.ui, 'update', () => {
+			if ( !getTableWidgetAncestor( viewDocument.selection ) ) {
+				this._hideView();
+			} else if ( this._isViewVisible ) {
+				repositionContextualBalloon( editor, 'table' );
+			}
+		} );
+
+		// Close on click outside of balloon panel element.
+		clickOutsideHandler( {
+			emitter: view,
+			activator: () => this._isViewInBalloon,
+			contextElements: [ this._balloon.view.element ],
+			callback: () => this._hideView()
+		} );
+
+		// Create the "UI -> editor data" binding.
+		// These listeners update the editor data (via table commands) when any observable
+		// property of the view has changed. This makes the view live, which means the changes are
+		// visible in the editing as soon as the user types or changes fields' values.
+		view.on( 'change:borderStyle', this._getPropertyChangeCallback( 'tableBorderStyle' ) );
+		view.on( 'change:borderColor', this._getPropertyChangeCallback( 'tableBorderColor' ) );
+		view.on( 'change:borderWidth', this._getPropertyChangeCallback( 'tableBorderWidth' ) );
+		view.on( 'change:backgroundColor', this._getPropertyChangeCallback( 'tableBackgroundColor' ) );
+		view.on( 'change:width', this._getPropertyChangeCallback( 'tableWidth' ) );
+		view.on( 'change:height', this._getPropertyChangeCallback( 'tableHeight' ) );
+		view.on( 'change:alignment', this._getPropertyChangeCallback( 'tableAlignment' ) );
+
+		return view;
+	}
+
+	/**
+	 * In this method the "editor data -> UI" binding is happening.
+	 *
+	 * When executed, this method obtains selected table property values from various table commands
+	 * and passes them to the {@link #view}.
+	 *
+	 * This way, the UI stays up–to–date with the editor data.
+	 *
+	 * @private
+	 */
+	_fillViewFormFromCommandValues() {
+		const commands = this.editor.commands;
+
+		this.view.set( {
+			borderStyle: commands.get( 'tableBorderStyle' ).value || DEFAULT_BORDER_STYLE,
+			borderColor: commands.get( 'tableBorderColor' ).value || '',
+			borderWidth: commands.get( 'tableBorderWidth' ).value || '',
+			backgroundColor: commands.get( 'tableBackgroundColor' ).value || '',
+			width: commands.get( 'tableWidth' ).value || '',
+			height: commands.get( 'tableHeight' ).value || '',
+			alignment: commands.get( 'tableAlignment' ).value || DEFAULT_ALIGNMENT,
```

Comment:
I'm quite puzzled about which default alignment should we use. Center? Or left? Or depending on the directionality of the content?

Also, the topic is broader than this. ATM changing the table alignment in the UI only changes the text alignment in all cells because our (block widget) styles enforce centered tables no matter the `alignment` DOM attribute value. It's a serious UX issue.

## Option 1 

We could make table widgets align their content to the left by default (technically, make sure they do nothing) and allow users to use the table properties UI to choose the alignment (I checked this, it works). But then... what would the tables look like if the properties plugin is not loaded? They should probably be centered. So this is a weird approach.

## Option 2

We could bind the alignment in the editing to the widget container class so:
* the DOM `alignment` of the `<table>` is still applied,
* the class of the widget container ensures visually everything looks right.

This class would be present only in editing. I wonder if this would actually work well (we downcast `<figure><table>...` so there's always a wrapper).

Reply:
I'll look into it more during the review - but it looks like something is wrong here - the table cells shouldn't be aligned left/right by the align table command. I didn't read the whole case though - so I might be missing something.

Reply:
That's a conversion bug - I don't know why the `'alignment'` attribute is downcasted here but probably the best way is to rename table attributes :/ So here it could be `tableAlignment` instead of `alignment`. I'll confirm this with an issue.

Reply:
Shouldn't table alignment should align **the entire table** to left/right/center? Not its content.

Reply:
OK, I can see that https://github.com/ckeditor/ckeditor5/issues/6179 clarifies that the current behaviour is a result of a bug.

Reply:
@Reinmar yes - details are here: https://github.com/ckeditor/ckeditor5/issues/6179.

Reply:
I'd move the conversion on this to linked issue (:point_up:) as the UI operates on commands and the bug is in editing part of the feature :/